### ISSUE TITLE
Add Markdown Language Grammar

### ISSUE THREAD

**Issue by bobrocke** [osxpeppermint/peppermint#52]
I use Markdown quite a bit for blogging and would like a syntax grammar so that it can be themed.

Markdown has a lot of variations; I'd like Markdown Extra which would cover the very popular GFM, too.

**Comment by osxpeppermint** [comment]
Returning to your suggestion... and in order to be 100% clear: 

Are you talking about something else other than the already-included Markdown Highlighting?

P.S. Btw, I've looked for "Markdown Extra" and I'm not sure I can find some specific grammar. Has it been implemented for any other editor? Sublime, TextMate or Atom?

**Comment by bobrocke** [comment]
Yes, something beyond the "standard" Markdown - whatever that is.

Markdown Extra would be a bonus. Perhaps GFM is widely enough used that would be a better target.

Anyway, I'm thinking that an enhancement to the current Markdown syntax might be the way to go. The improved grammar could cover Gruber's Markdown and add in GFM (and maybe other dialects in time).

**Comment by osxpeppermint** [comment]
OK, I'll have a look into it! :)
